Kernel/Output/HTML/Preferences/ColumnFilters.pm

```diff
@@ -57,14 +59,28 @@ sub Run {
 
     return 1 if !defined $FilterAction;
 
-    for my $Key ( sort keys %{ $Param{GetParam} } ) {
+    if ( !$Kernel::OM->Get('Kernel::Config')->Get('DemoSystem') ) {
+        KEY:
+        for my $Key ( sort keys %{ $Param{GetParam} } ) {
 
-        # pref update db
-        if ( !$Kernel::OM->Get('Kernel::Config')->Get('DemoSystem') ) {
+            # We require non-empty list of column names - ignore antyhing else.
+            next KEY if !( $Param{GetParam}->{$Key} && IsArrayRefWithData( $Param{GetParam}->{$Key} ) );
+
+            # Remove column name duplicates from list preserving column order.
+            my %SeenColumnNames;
+            my @DeduplicatedColumnNameArray;
+            COLUMN_NAME:
+            for my $ColumnName ( @{ $Param{GetParam}->{$Key} } ) {
+                next COLUMN_NAME if $SeenColumnNames{$ColumnName};
+                $SeenColumnNames{$ColumnName} = 1;
+                push @DeduplicatedColumnNameArray, $ColumnName;
+            }
+
+            # Update list in DB.
             $Kernel::OM->Get('Kernel::System::User')->SetPreferences(
                 UserID => $Param{UserData}->{UserID},
                 Key    => $Key . '-' . $FilterAction,
-                Value  => $Kernel::OM->Get('Kernel::System::JSON')->Encode( Data => $Param{GetParam}->{$Key} ),
+                Value  => $Kernel::OM->Get('Kernel::System::JSON')->Encode( Data => \@DeduplicatedColumnNameArray ),
```

Comment:
```suggestion
                Value  => $Kernel::OM->Get('Kernel::System::JSON')->Encode( Data => \@UniqueColumnNames ),
```

Reply:
Done in 4252343ca6f6cdf84d540a9c08635c414511d75b.